src/modules/share/share.repository.ts

```diff
@@ -238,6 +243,57 @@ export class SequelizeShareRepository implements ShareRepository {
     await this.shareModel.destroy({ where: { id: shareId } });
   }
 
+  async findAllSharedFoldersToAUserPaginated(
+    user: User,
+    page: number,
+    perPage: number,
+    orderBy?: OrderBy,
+  ): Promise<Folder[]> {
+    const { offset, limit } = Pagination.calculatePagination(page, perPage);
+
+    const validTypePaginationParameters =
+      (isNaN(page) && !isNaN(perPage)) || (!isNaN(page) && isNaN(perPage));
+
+    if (validTypePaginationParameters) {
+      throw new BadRequestException('take and skip must be provided together');
+    }
+
+    const order = orderBy
+      ? [orderBy.split(':') as [string, string]]
+      : undefined;
+
+    const findCriteria = {
+      where: {
+        user_id: user.id,
+        is_folder: true,
+      },
+      include: [
+        {
+          model: this.folderModel,
+          where: {
+            deleted: false,
+          },
+          required: false,
+        },
+      ],
+    };
+
+    if (!isNaN(limit) && !isNaN(offset)) {
```

Comment:
The repository layer should persist data only, the responsibility of validating the data should be done by the controller. All the controllers must validate the data so any usecase/repository receives the data in the expected format. If we have types for variables but we still have to validate things on the entire codebase, then TypeScript is not worth it